allpairspy/allpairs.py

```diff
@@ -180,15 +181,15 @@ def __resort_working_array(self, chosen_item_list, num):
                 for i in range(0, self.__n)
             ]
 
-            # weighting the node node that creates most of new pairs is the best
+            # weighting the node that creates most of new pairs is the best
             weights = [-len(new_combs[-1])]
 
             # less used outbound connections most likely to produce more new
             # pairs while search continues
             weights.extend(
                 [len(data_node.out)]
                 + [len(x) for x in reversed(new_combs[:-1])]
-                + [-data_node.counter]  # less used node is better
+                + [data_node.counter]  # less used node is better
```

Comment:
This fixes uneven distribution of parameters.

If you want less used nodes to be first after resorting, then you shouldn't negate node's usage number. I believe it was just a typo here.

Reply:
As you mentioned `-len(data_node.in_)` would be get better results.
Could you update to using `-len(data_node.in_)`?

Reply:
Sure, but I'll be able to do it in a week from now.